Local-devcontainers (#2662)
dotnet/docs-aspire: docs/get-started/dev-containers.md

1. [Create a new repository](https://github.com/new?template_name=aspire-devcontainer&template_owner=dotnet) using our template.

    :::image source="media/new-repository-from-template.png" lightbox="media/new-repository-from-template.png" alt-text="Create new repository.":::

    Once you provide the details and select **Create repository**, the repository is created and shown in GitHub.

1. Clone the repository to your local developer workstation using the following command:

```dotnetcli
git clone https://github.com/<org/username>/<repository>
```

1. Open the repository in Visual Studio Code. After a few moments Visual Studio Code will detect the _.devcontainer/devcontainer.json_ file and prompt to open the repository inside a container. Select whichever option is most appropriate for your workflow.

![Prompt to open repository inside a container](media/reopen-in-container.png)

After a few moments the list of files will become visible and the local build of the dev container will be completed.

![Devcontainer build completed](media/devcontainer-build-completed.png)

> 1. Open a new terminal window in VS Code <kbd>Ctrl</kbd>+(<kbd>Shift</kbd>+<kbd>`</kbd>) and create a new .NET Aspire project using the `dotnet` command-line.

Reviewer: ```suggestion
1. Open a new terminal window in VS Code <kbd>Ctrl</kbd>+(<kbd>Shift</kbd>+<kbd>\`</kbd>) and create a new .NET Aspire project using the `dotnet` command-line.
```

```dotnetcli
dotnet new aspire-starter -n HelloAspire
```

After a few moments the project will be created and initial dependencies restored.

> 5. Open the `ProjectName.AppHost/Program.cs` file in the editor and click the run button on the top right hand corner of the editor window.

Reviewer: ```suggestion
1. Open the _ProjectName.AppHost/Program.cs_ file in the editor and select the run button on the top right corner of the editor window.
```

![Run button in editor](media/vscode-run-button.png)

Visual Studio Code will build and start the .NET Aspire AppHost and will automatically open the .NET Aspire Dashboard. Because the endpoints hosted in the container are using a self-signed certificate the first time you access an endpoint for a specific devcontainer you will be presented with a certificate error.

![Browser certificate error](media/browser-certificate-error.png)

This is expected. Once you have confirmed that the URL being requested corresponds to the dashboard in the Devcontainer you can ignore this warning.

![.NET Aspire dashboard running in Devcontainer](media/aspire-dashboard-in-devcontainer.png)

.NET Aspire will automatically configure forwarded ports so that when you click on the endpoints in the .NET Aspire dashboard they will be tunneled to processes and nested containers within the Devcontainer.

> 6. Commit changes to the GitHub repository

Reviewer: ```suggestion
1. Commit changes to the GitHub repository
```
